packages/remix-dev/devServer_unstable/hdr.ts

```diff
@@ -71,11 +71,23 @@ export let detectLoaderChanges = async (
             );
 
             let contents = "module.exports = {};";
-            if (theExports.includes("loader")) {
+            if (
+              theExports.includes("loader") &&
+              theExports.includes("clientLoader")
+            ) {
+              contents = `export { loader, clientLoader } from ${JSON.stringify(
+                `./${file}`
+              )};`;
+            } else if (theExports.includes("loader")) {
               contents = `export { loader } from ${JSON.stringify(
                 `./${file}`
               )};`;
+            } else if (theExports.includes("clientLoader")) {
+              contents = `export { clientLoader } from ${JSON.stringify(
+                `./${file}`
+              )};`;
```

Comment:
Enable HDR for clientLoader by adapting the current tree shaking approach for fine-grained `loader`-change detection to also support `clientLoader`.  Need these branches since `esbuild` will complain if we try to import a non-existent named export